orchagent/interface_rates.lua

```diff
@@ -0,0 +1,90 @@
+-- KEYS - rif IDs
+-- ARGV[1] - counters db index
+-- ARGV[2] - counters table name
+-- ARGV[3] - poll time interval
+-- return nothing
+
+local asic_db = 1 
+local counters_db = ARGV[1]
+local config_db = 4 
+local counters_table_name = ARGV[2] 
+-- local asic_table_name = "ASIC_STATE:SAI_OBJECT_TYPE_PORT" 
+local rates_table_name = "RATES"
+
+-- Get configuration
+redis.call('SELECT', config_db)
+local smooth_interval = redis.call('GET', rates_table_name .. ':' .. 'RIF_SMOOTH_INTERVAL')
+local alpha = redis.call('GET', rates_table_name .. ':' .. 'RIF_ALPHA')
+local one_minus_alpha = 1 - alpha
+local delta = tonumber(ARGV[3])
+
+-- Get speeds
+local speeds = {}
+redis.call('SELECT', asic_db)
+local n = table.getn(KEYS)
+for i = 1, n do
+    speeds[i] = 40000 
+
+redis.call('SELECT', counters_db)
+
+local initialized = redis.call('GET', rates_table_name .. ':' .. 'INIT_DONE')
+
+for i = 1, n do
+    -- Get new COUNTERS values
+    local in_octets = redis.call('HGET', counters_table_name .. ':' .. KEYS[i], 'SAI_ROUTER_INTERFACE_STAT_IN_OCTETS')
+    local in_pkts = redis.call('HGET', counters_table_name .. ':' .. KEYS[i], 'SAI_ROUTER_INTERFACE_STAT_IN_PACKETS')
+    local out_octets = redis.call('HGET', counters_table_name .. ':' .. KEYS[i], 'SAI_ROUTER_INTERFACE_STAT_OUT_OCTETS')
+    local out_pkts = redis.call('HGET', counters_table_name .. ':' .. KEYS[i], 'SAI_ROUTER_INTERFACE_STAT_OUT_PACKETS')
+
+    if initialized == "DONE" or initialized == "COUNTERS_LAST" then
+        -- Get old COUNTERS values
+        local in_octets_pkts_last = redis.call('HGET', rates_table_name .. ':' .. KEYS[i], 'SAI_ROUTER_INTERFACE_STAT_IN_OCTETS_last')
+        local in_pkts_last = redis.call('HGET', rates_table_name .. ':' .. KEYS[i], 'SAI_ROUTER_INTERFACE_STAT_IN_PACKETS_last')
```

Comment:
What will be behavior if this SAI Attribute is not supported by given ASIC ?  Will there be any error or rate will default to 0. ?

Reply:
+1, had the same question for this + the other lua script.

Reply:
The script will exit with error if the counter is unavailable in db